MekHQ/src/mekhq/campaign/autoresolve/acar/handler/StandardUnitAttackHandler.java

```diff
@@ -0,0 +1,231 @@
+            targetUnit.addTargetingCrit();
+            reporter.reportTargetingCrit(targetUnit);
+        } else if (criticalRollResult <= 9) {
+            // Damage crit
+            targetUnit.addDamageCrit();
+            reporter.reportDamageCrit(targetUnit);
+        } else if (criticalRollResult <= 11) {
+            // Both targeting and damage crit
+            targetUnit.addTargetingCrit();
+            targetUnit.addDamageCrit();
+            reporter.reportTargetingCrit(targetUnit);
+            reporter.reportDamageCrit(targetUnit);
+        } else {
+            // Destroyed
+            countKill(attackingUnit, targetUnit);
+            targetUnit.setCurrentArmor(0);
+            target.setHighStressEpisode();
+            reporter.reportUnitDestroyed();
+        }
+    }
+
+
+    private void countKill(SBFUnit attackingUnit, SBFUnit targetUnit) {
+
+        var killers = attackingUnit.getElements().stream().map(AlphaStrikeElement::getId)
+            .map(e -> simulationManager().getGame().getEntity(e)).filter(Optional::isPresent).map(Optional::get).toList();
+        var targets = targetUnit.getElements().stream().map(AlphaStrikeElement::getId)
+            .map(e -> simulationManager().getGame().getEntity(e)).filter(Optional::isPresent).map(Optional::get).toList();
+        for (var target : targets) {
+            RandomUtils.fastSample(killers).ifPresent(e -> e.addKill(target));
+        }
+    }
+
+    private int processDamageByEngagementControl(Formation attacker, Formation target, int damage) {
+        var engagementControlMemories = attacker.getMemory().getMemories("engagementControl");
+        var engagement = engagementControlMemories
+            .stream()
+            .filter(f -> f.getOrDefault("targetFormationId", Entity.NONE).equals(target.getId()))
+            .findFirst(); // there should be only one engagement control memory for this target
+
+        if (engagement.isEmpty()) {
+            return damage;
+        }
+
+        var isAttacker = (boolean) engagement.get().getOrDefault("attacker", false);
+
+        if (!isAttacker) {
+            return damage;
+        }
+
+        var wonEngagement = (boolean) engagement.get().getOrDefault("wonEngagementControl", false);
+        var engagementControl = (EngagementControl) engagement.get().getOrDefault("engagementControl", EngagementControl.NONE);
+        if (wonEngagement) {
+            return processDamageEngagementControlVictory(engagementControl, damage);
+        }
+        return processDamageEngagementControlDefeat(engagementControl, damage);
+    }
+
+    private int processDamageEngagementControlDefeat(EngagementControl engagementControl, double damage) {
+        switch(engagementControl) {
+            case NONE:
+                // nothing happens
+                break;
+            case OVERRUN:
+                break;
+            case STANDARD:
+                break;
+            case EVADE:
+                damage = damage * 0.5;
```

Comment:
Actually, on second thoughts, seeing that you only have one case where something happens, would it not be better as an if-statement?

Reply:
fixing